libraries/launcher/org/prismlauncher/log4jinjector/UniversalLog4JInjector.java

```diff
@@ -0,0 +1,41 @@
+package org.prismlauncher.log4jinjector;
+
+import java.util.logging.Logger;
+import java.util.logging.Level;
+
+public class UniversalLog4JInjector {
+    private static final Logger LOGGER = Logger.getLogger("UniversalLog4JInjector");
+
+    public static void inject() {
+        if (System.getProperty("prism.log4j.inject").equals("false")) {
+            return;
+        }
+        String l4jVersion;
+        try {
+            l4jVersion = Class.forName("org.apache.logging.log4j.LogManager").getPackage().getImplementationVersion();
+            assert l4jVersion != null;
+        } catch (Exception e) {
+            LOGGER.log(Level.WARNING, "Log4J not found. Is this Minecraft using Log4J?");
```

Comment:
This will create a warning on Minecraft versions that don't use Log4j which isn't great.